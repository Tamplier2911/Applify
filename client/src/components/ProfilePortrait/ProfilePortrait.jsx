// import './ProfilePortrait.scss';
import React from "react";

// components
import AdminBar from "../AdminBar/AdminBar";

// JS Rendering CSS
import {
  ProfilePortraitContainer,
  ProfilePortraitWrapper,
  ProfilePortraitImage,
} from "./ProfilePortraitStyles";

const ProfilePortrait = ({ photo, userRole }) => {
  return (
    <ProfilePortraitContainer animateIn="bounceInRight" animateOnce={true}>
      <ProfilePortraitWrapper>
        <ProfilePortraitImage src={photo} alt="happy user" />
      </ProfilePortraitWrapper>
      {userRole === "admin" || userRole === "owner" ? <AdminBar /> : null}
    </ProfilePortraitContainer>
  );
};

export default ProfilePortrait;
