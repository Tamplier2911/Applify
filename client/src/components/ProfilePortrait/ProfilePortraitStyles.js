import styled from "styled-components";

export const ProfilePortraitContainer = styled.div`
  display: grid;
  box-shadow: 0 0.1rem 0.5rem var(--cl-font);
  padding: 2rem;
`;

export const ProfilePortraitWrapper = styled.div`
  justify-self: center;
  align-self: center;

  width: 20rem;
  height: 20rem;
  border-radius: 50%;
  box-shadow: 0 0.2rem 0.5rem var(--cl-shadow);
  overflow: hidden;

  @media only screen and (max-width: 768px) {
    width: 14rem;
    height: 14rem;
    // border-radius: 0%;
  }
`;

export const ProfilePortraitImage = styled.img`
  width: 100%;
  height: 100%;
  display: block;
  object-fit: cover;
`;
