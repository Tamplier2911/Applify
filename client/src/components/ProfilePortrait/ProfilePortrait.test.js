import React from "react";
import toJson from "enzyme-to-json";

// component
import ProfilePortrait from "./ProfilePortrait";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<ProfilePortrait />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<ProfilePortrait photo="user.jpg" userRole="user" />);
  });

  it("renders without an error", () => {
    const profilePortrait = findByTestAttr(wrapper, "profile-portrait");
    expect(profilePortrait).toHaveLength(1);
  });

  it("renders children - user portrait", () => {
    const profilePortraitPhoto = findByTestAttr(
      wrapper,
      "profile-portrait-photo"
    );
    expect(profilePortraitPhoto).toHaveLength(1);
  });

  it("renders children - admin bar if userRole is admin or owner", () => {
    wrapper = setShallow(<ProfilePortrait photo="user.jpg" userRole="admin" />);
    const profilePortraitAdminBar = findByTestAttr(
      wrapper,
      "profile-portrait-adminbar"
    );
    expect(profilePortraitAdminBar).toHaveLength(1);
  });

  it("recieves expected props", () => {
    const expectedProps = { photo: "user.jpg", userRole: "user" };
    const err = checkProps(ProfilePortrait, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const profilePortrait = findByTestAttr(wrapper, "profile-portrait");
    expect(toJson(profilePortrait)).toMatchSnapshot();
  });
});
