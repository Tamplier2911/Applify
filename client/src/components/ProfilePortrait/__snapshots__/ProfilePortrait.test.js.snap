// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePortrait /> matches snapshot 1`] = `
<Styled(ScrollAnimation)
  animateIn="bounceInRight"
  animateOnce={true}
  data-test="profile-portrait"
>
  <styled.div
    data-test="profile-portrait-photo"
  >
    <styled.img
      alt="happy user"
      src="user.jpg"
    />
  </styled.div>
</Styled(ScrollAnimation)>
`;
